mne/viz/_3d.py

```diff
@@ -733,24 +764,29 @@ def plot_source_estimates(stc, subject=None, surface='inflated', hemi='lh',
         hemis = [hemi]
 
     title = subject if len(hemis) > 1 else '%s - %s' % (subject, hemis[0])
-    args = _get_args(Brain.__init__)
-    kwargs = dict(title=title, figure=figure, config_opts=config_opts,
-                  subjects_dir=subjects_dir)
-    if 'views' in args:
-        kwargs['views'] = views
     with warnings.catch_warnings(record=True):  # traits warnings
-        brain = Brain(subject, hemi, surface, **kwargs)
+        brain = Brain(subject, hemi, surface, True, title, cortex, size,
+                      background, foreground, figure, subjects_dir, views,
+                      config_opts=config_opts)
+
+    if time_unit == 's':
+        times = stc.times
+    elif time_unit == 'ms':
+        times = 1e3 * stc.times
+    else:
+        raise ValueError("time_unit needs to be 's' or 'ms', got %s" %
+                         repr(time_unit))
```

Comment:
This check should go toward the top -- checking parameters as early as possible is best, plus I think it can be combined a bit with the other ways that it time units get triaged above